homeassistant/components/upnp/device.py

```diff
@@ -11,9 +12,8 @@
 from async_upnp_client.client_factory import UpnpFactory
 from async_upnp_client.exceptions import UpnpError
 from async_upnp_client.profiles.igd import IgdDevice, StatusInfo
+from getmac import get_mac_address
```

Comment:
This isn't reliable and won't work in all environments.  We can only use it to improve existing behavior, but need to downgrade gracefully when it is not usable.

Reply:
Alright. Do you think using the IP, like it used to do so before, is reliable?

Reply:
It is already using the UDN of the device. The mac address is already a fall back. I'd say, lets try it, and if needed, the device IP can be added as well.